src/TabSwipe/widget/SwipeHandler.ts

```diff
@@ -0,0 +1,136 @@
+        this.hammer.on("panend", event => this.onPanEnd(event));
+
+        this.loadPanes();
+    }
+
+    private loadPanes() {
+        const activePane = this.tabContainer._active;
+        if (!this.lazyLoad) {
+            this.tabPanes.forEach(tabPane => {
+                this.showTab(tabPane, true);
+            });
+        } else {
+            this.tabPanes.forEach(tabPane => {
+                this.setTabPanePosition(tabPane);
+            });
+        }
+        this.showTab(activePane, true);
+    }
+
+    private showTab(activePane: TabPane, updatePosition = false) {
+        this.tabContainer.showTab(activePane);
+        if (updatePosition) {
+            this.setTabPanePosition(activePane);
+        }
+    }
+
+    private onPanStart(event: HammerInput) {
+        if (event.pointerType === "mouse") return;
+        this.activePane.visibilityIndex = this.visibleTabs.indexOf(this.activePane);
+        this.panStartPosition = event.deltaX;
+        this.isSwiping = true;
+    }
+
+    private onPanMove(event: HammerInput) {
+        if (this.isSwiping && this.authorizeSwipe(event)) {
+            this.lazyLoadTabPane(event);
+            this.tabContainerContent.classList.remove("animate");
+            this.tabContainerContent.style.transform = `translate3d(${this.getPercentageMoved(event)}%, 0, 0)`;
+        } else {
+            this.isSwiping = false;
+        }
+    }
+
+    private onPanEnd(event: HammerInput) {
+        if (this.isSwiping) {
+            this.isSwiping = false;
+            this.dependentPaneVisible = false;
+            const percentageMoved = this.getPercentageMoved(event);
+            if (Math.abs(percentageMoved) > this.threshold) {
+                const newActiveIndex = this.activePane.visibilityIndex + this.getDirection(event);
+                this.tabContainer.showTab(this.visibleTabs[ newActiveIndex ]);
+            }
+            this.tabContainerContent.classList.add("animate");
+            this.tabContainerContent.style.transform = `translate3d(0, 0, 0)`;
+        }
+    }
+
+    private lazyLoadTabPane(event: HammerInput) {
+        if (this.lazyLoad && this.isSwiping && !this.dependentPaneVisible) {
+            this.showTab(this.tabPanes[this.activePane.visibilityIndex + this.getDirection(event)]);
+            this.showTab(this.activePane);
+            this.dependentPaneVisible = true;
+        }
+    }
+
+    private authorizeSwipe(event: HammerInput): boolean {
+        return (this.getDirection(event) > 0 && this.activePane.visibilityIndex < this.visibleTabs.length - 1)
+            || (this.getDirection(event) < 0 && this.activePane.visibilityIndex !== 0);
+    }
+
+    private getPercentageMoved(event: HammerInput): number {
+        return (100 / this.containerWidth) * (event.deltaX - this.panStartPosition);
+    }
+
+    private getDirection(event: HammerInput): number {
+        return (event.deltaX - this.panStartPosition) > 0 ? -1 : 1;
```

Comment:
it can never be zero since hammer pan events are fired after a threshold of 10px delta movement